bitbots_vision_tools/scripts/colorpicker.py

```diff
@@ -0,0 +1,254 @@
+    """
+    def __init__(self):
+        rospy.init_node('bitbots_colorpicker')
+        rospy.loginfo('Initializing colorpicker...', logger_name="colorpicker")
+        rospack = rospkg.RosPack()
+        self._package_path = rospack.get_path('bitbots_vision_tools')
+        self._cv_bridge = CvBridge()
+        # Load default image
+        self._image = cv2.imread(os.path.join(self._package_path, "img", "Colorpicker.png"))
+
+        # Subscribe to Image-message
+        self._sub_image_msg = rospy.Subscriber(
+            "image_raw",
+            Image,
+            self._image_callback,
+            queue_size=1,
+            buff_size=60000000,
+            tcp_nodelay=False)
+
+        # Init
+        self._init_text()
+        cv2.namedWindow("Colorpicker")
+        cv2.setMouseCallback("Colorpicker", self._mouse_callback)
+        self._mouse_coord = (0,0)
+        self._left_click, self._undo_click = False, False
+        self._default_path = "~/.ros/color_space.pickle"
+        color_space = np.zeros((256, 256, 256), dtype=np.uint8)
+
+        # Default box size
+        self._box_size = 50
+
+        # Init history
+        self._history = [color_space]
+
+
+    def _mouse_callback(self, event, x, y, flags, param):
+        """
+        Callback for the OpenCV cursor.
+
+        :param event: Event type
+        :param x: Mouse x position
+        :param y: Mouse y position
+        :param flags: Callback flags
+        :param param: Some unused parameter
+        """
+        self._undo_click = (flags == (cv2.EVENT_FLAG_SHIFTKEY + cv2.EVENT_FLAG_LBUTTON))
+        self._left_click = (not self._undo_click and event == cv2.EVENT_LBUTTONUP)
+
+        # Set self._mouse_coordinates
+        self._mouse_coord = (x, y)
+
+
+    def _draw_mask(self, image, mask, color, opacity=0.5):
+        """
+        Draws the mask on an image
+
+        :param image: The image canvas
+        :param mask: The binary mask
+        :param color: Color bgr tuple
+        :param opacity: Opacity of the mask
+        """
+        # Make a colored image
+        colored_image = np.zeros_like(image)
+        colored_image[:, :] = tuple(np.multiply(color, opacity).astype(np.uint8))
+
+        # Compose debug image
+        return cv2.add(cv2.bitwise_and(image, image, mask=255-mask),
+                cv2.add(colored_image*opacity, image*(1-opacity), mask=mask).astype(np.uint8))
+
+
+    def _init_text(self):
+        """
+        Prints some usage information.
+        """
+        rospy.loginfo(
+            "Welcome to the Colorpicker.\n"
+            "Click to select the colors in the OpenCV window.\n"
+            "Use <shift> and <left click> to remove colors.\n"
+            "To undo a step press 'u'.\n"
+            "To increase the size of the selection box press '+', to decrease press '-' \n"
+            "To save the color space press 's' followed by the path and <enter>.\n"
+            "Exit using <esc>.\n\n\n", logger_name="colorpicker")
```

Comment:
Maybe remove these many `\n` and instead add some spacers like `###########` before and after the _content_ lines.

Reply:
I think the blank lines clean the console a bit.